server/src/diagnostic_ext.rs

```diff
@@ -28,24 +29,33 @@ pub async fn publish_diagnostics_for_set(
         .filter_map(|uri| Some((convert_slice_url_to_uri(uri)?, vec![])))
         .collect::<HashMap<Url, Vec<tower_lsp::lsp_types::Diagnostic>>>();
 
-    // Process the diagnostics and populate the map
-    process_diagnostics(diagnostics, &mut map);
+    // Process the diagnostics and populate the map. Any diagnostics that do not have a span are returned for further processing.
+    let spanless_diagnostics = process_diagnostics(diagnostics, &mut map);
+    for diagnostic in spanless_diagnostics.iter() {
+        backend
```

Comment:
This doesn't wait dismissal, just the sending to the client?

Reply:
Correct, `show_popup` just calls `send_notification` to the client which requires awaiting. Therefore for each diagnostic we are just waiting for the notification to be sent.